File: fixtures/scale/kotlin/large/src/cryptobench/registry/HashRegistry.kt
package cryptobench.registry

import cryptobench.core.CipherSuite
import cryptobench.ciphers.fnv1a.Fnv1aSuite
import cryptobench.ciphers.djb2.Djb2Suite
import cryptobench.ciphers.sdbm.SdbmSuite
import cryptobench.ciphers.adler32.Adler32Suite
import cryptobench.ciphers.crc32.Crc32Suite
import cryptobench.ciphers.fletcher.FletcherSuite
import cryptobench.ciphers.pearson.PearsonSuite

object HashRegistry {
    fun suites(): List<CipherSuite> = listOf(
        Fnv1aSuite(),
        Djb2Suite(),
        SdbmSuite(),
        Adler32Suite(),
        Crc32Suite(),
        FletcherSuite(),
        PearsonSuite(),
    )
}

File: fixtures/scale/dart/large/lib/registry/hash_registry.dart
import '../core/cipher_suite.dart';
import '../ciphers/fnv1a/fnv1a_suite.dart';
import '../ciphers/djb2/djb2_suite.dart';
import '../ciphers/sdbm/sdbm_suite.dart';
import '../ciphers/adler32/adler32_suite.dart';
import '../ciphers/crc32/crc32_suite.dart';
import '../ciphers/fletcher/fletcher_suite.dart';
import '../ciphers/pearson/pearson_suite.dart';

List<CipherSuite> hashSuites() => [
  Fnv1aSuite(),
  Djb2Suite(),
  SdbmSuite(),
  Adler32Suite(),
  Crc32Suite(),
  FletcherSuite(),
  PearsonSuite(),
];
